src/main/java/org/opensearch/neuralsearch/query/HybridBulkScorer.java

```diff
@@ -179,13 +179,19 @@ private boolean allDocIdsUsed(int[] docsIds, int max) {
     }
 
     private int getNextDocIdCandidate(final int[] docsIds) {
-        int nextDoc = -1;
-        for (int doc : docsIds) {
-            if (doc != DocIdSetIterator.NO_MORE_DOCS) {
-                nextDoc = Math.max(nextDoc, doc);
+        int[] nextDoc = new int[docsIds.length];
+        Arrays.fill(nextDoc, -1);
+        int minDocIdForNextIteration = Integer.MAX_VALUE;
+        for (int i = 0; i < nextDoc.length; i++) {
+            if (docIds[i] != DocIdSetIterator.NO_MORE_DOCS) {
+                nextDoc[i] = Math.max(nextDoc[i], docIds[i]);
+                if (nextDoc[i] < minDocIdForNextIteration) {
+                    minDocIdForNextIteration = nextDoc[i];
+                }
             }
         }
-        return nextDoc == -1 ? DocIdSetIterator.NO_MORE_DOCS : nextDoc;
+
+        return minDocIdForNextIteration == -1 ? DocIdSetIterator.NO_MORE_DOCS : minDocIdForNextIteration;
```

Comment:
is this scenario even possible: `minDocIdForNextIteration == -1`? I think this is dead code now